enhancements/machine-config/mco-cgroupsv2-support.md

```diff
@@ -0,0 +1,121 @@
+title: CGroupsv2 Machine Config Pool Enablement
+authors:
+  - "@rphillips"
+reviewers:
+  - "@mrunalp"
+approvers:
+  - "@yuqi-zhang"
+  - "@mrunalp"
+creation-date: 2021-10-19
+last-updated: 2021-10-19
+status: implementable
+---
+
+# CgroupsV2 Machine Config Pool Enablement
+
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+cgroupsv2 enablement in Kubernetes has progressed to the point that it is
+usable. The underlying runtime (cri-o) and supporting subsystems are now ready
+for customers to begin their own testing with it. Not all workloads will be
+compatible with cgroupsv2, so it will *not* be enabled by default within
+Openshift at this time.
+
+## Motivation
+
+Migrating to Cgroupsv2 will bring in many new features and fixes not found in
+Cgroupsv1. Cgroupsv1 is considered 'legacy' and migrating to Cgroupsv2 is
+considered necessary since RHEL ships with Cgroupsv2 on by default. (Openshift
+4.x currently disables Cgroupsv2 in favor of Cgroupsv1).
+
+Some features of Cgroupsv2 include:
+
+* IO enhancments
+* User based OOM killer
+* Cgroup aware OOM killer
+
+[Kubernetes On Cgroupsv2 - Video](https://www.youtube.com/watch?v=u8h0e84HxcE)
+
+### Goals
+
+- [ ] Enable Cgroupsv2 on a Machine Config Pool
+
+### Non-Goals
+
+## Proposal
+
+[Machine Config
+Pools](https://github.com/openshift/machine-config-operator/blob/85f9f9583451cb1e4c25893598eb5d086fd7e4ba/pkg/apis/machineconfiguration.openshift.io/v1/types.go#L191)
+have a FIPS flag on them to enable `FIPS` within the system. Adding a `Cgroupsv2
+bool` flag to the Machine Config Pool, when true would enable Cgroupsv2 on the
+pool.
+
+The installer passes down the `FIPS` flag to MCO on bootstrap. Similar
+functionality can be added to pass down a `Cgroupsv2` flag at install to enable
+a Cgroupsv2 only cluster.
+
+The following kernel command line arguments would be set when `Cgroupsv2` is set to
+true:
+```yaml
+  kernelArguments:
+    - systemd.unified_cgroup_hierarchy=1
+    - cgroup_no_v1="all"
+    - psi=1 
+```
+
+### User Stories
+
+#### Story 1
+
+Add Cgroupsv2 flag to Machine Config Pool type
+
+This story would also require wiring up the kernel commandline options to enable
+or disable Cgroupsv2 on the necessary pool.
+
+Acceptance Criteria
+- Unit Test
+- Integration Test
+
+### Risks and Mitigations
+
+The primary risk of Cgroupsv2 is some workloads not supporting the changed
+cgroup options. This is the reason why it is off by default.
+
+## Design Details
+
+### Test Plan
+
+Testing should be thoroughly done at all levels, including unit, end-to-end, and
+integration.
+
+### Graduation Criteria
+
+#### Tech Preview -> GA
```

Comment:
What are the actual graduation criteria for tech preview? ie what do you expect to deliver

Reply:
Updated with graduation requirements